datascience/util.py

```diff
@@ -53,3 +54,47 @@ def plot_cdf_area(rbound, lbound=None, mean=0, sd=1):
     plt.title("Normal Curve ~ ($\mu$ = {0}, $\sigma$ = {1}) "
               "{2} < z < {3}".format(mean, sd, llabel, rbound), fontsize=16)
     plt.show()
+
+
+def table_apply(table, func, subset=None):
+    """Applies a function to each column and returns a Table.
+
+    Uses pandas `apply` under the hood, then converts back to a Table
+
+    Parameters
+    ----------
+    table : instance of Table
+        The table to apply your function to
+    func : function
+        Any function that will work with DataFrame.apply
+    subset : list | None
+        A list of columns to apply the function to. If None, function
+        will be applied to all columns in table
+
+    Returns
+    -------
+    tab : instance of Table
+        A table with the given function applied. It will either be the
+        shape == shape(table), or shape (1, table.shape[1])
+    """
+    import pandas as pd
+    from datascience import Table
```

Comment:
how does that work? just `try: from . import Table; except ImportError:
somethingsomething`? Anywhere in the code where you guys already do this so
I can just copy it?

On Tue, Jan 12, 2016 at 9:36 PM, Sam Lau <notifications@github.com> wrote:

> In datascience/util.py
> <https://github.com/dsten/datascience/pull/175#discussion_r49553209>:
>
> > +    table : instance of Table
> > +        The table to apply your function to
> > +    func : function
> > +        Any function that will work with DataFrame.apply
> > +    subset : list | None
> > +        A list of columns to apply the function to. If None, function
> > +        will be applied to all columns in table
> > +
> > +    Returns
> > +    -------
> > +    tab : instance of Table
> > +        A table with the given function applied. It will either be the
> > +        shape == shape(table), or shape (1, table.shape[1])
> > +    """
> > +    import pandas as pd
> > +    from datascience import Table
>
> Yeah — I think what we decided is to wrap imports in a try block so we
> can still use readthedocs.
>
> Also, we're going to be using pandas more for the Table class anyway so
> it doesn't benefit us very much to leave it here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dsten/datascience/pull/175/files#r49553209>.
>